From 26b83dfe7d383248100495fd7694295fa028cbe0 Mon Sep 17 00:00:00 2001
From: Daanturo <daanturo@gmail.com>
Date: Fri, 17 Jun 2022 18:02:17 +0700
Subject: [PATCH 1/3] Workaround GTK does not recognize Super modifier

---
 src/pgtkterm.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/pgtkterm.c b/src/pgtkterm.c
index 13f6c6c3c4..578acbda6f 100644
--- a/src/pgtkterm.c
+++ b/src/pgtkterm.c
@@ -5066,7 +5066,7 @@ pgtk_gtk_to_emacs_modifiers (struct pgtk_display_info *dpyinfo, int state)
     mod |= mod_ctrl;
   if (state & GDK_META_MASK || state & GDK_MOD1_MASK)
     mod |= mod_meta;
-  if (state & GDK_SUPER_MASK)
+  if (state & (GDK_SUPER_MASK | GDK_MOD4_MASK))
     mod |= mod_super;
   if (state & GDK_HYPER_MASK)
     mod |= mod_hyper;
@@ -5089,7 +5089,7 @@ pgtk_emacs_to_gtk_modifiers (struct pgtk_display_info *dpyinfo, int state)
 
   mask = 0;
   if (state & mod_super)
-    mask |= GDK_SUPER_MASK;
+    mask |= (GDK_SUPER_MASK | GDK_MOD4_MASK);
   if (state & mod_hyper)
     mask |= GDK_HYPER_MASK;
   if (state & shift_modifier)
@@ -5207,7 +5207,7 @@ key_press_event (GtkWidget *widget, GdkEvent *event, gpointer *user_data)
       /* While super is pressed, the input method will always always
 	 resend the key events ignoring super.  As a workaround, don't
 	 filter key events with super or hyper pressed.  */
-      if (!(event->key.state & (GDK_SUPER_MASK | GDK_HYPER_MASK)))
+ if (!(event->key.state & ((GDK_SUPER_MASK | GDK_MOD4_MASK) | GDK_HYPER_MASK)))
 	{
 	  if (pgtk_im_filter_keypress (f, &event->key))
 	    return TRUE;
@@ -5223,7 +5223,7 @@ key_press_event (GtkWidget *widget, GdkEvent *event, gpointer *user_data)
          Don't do it here because XLookupString is too eager.  */
       state &= ~GDK_CONTROL_MASK;
       state &= ~(GDK_META_MASK
-		 | GDK_SUPER_MASK | GDK_HYPER_MASK | GDK_MOD1_MASK);
+		 | (GDK_SUPER_MASK | GDK_MOD4_MASK) | GDK_HYPER_MASK | GDK_MOD1_MASK);
 
       nbytes = event->key.length;
       if (nbytes > copy_bufsiz)
-- 
2.38.1


From fe9f0f225298ee9fa2f0d67dd132a58a73992b3a Mon Sep 17 00:00:00 2001
From: Daanturo <daanturo@gmail.com>
Date: Tue, 22 Nov 2022 14:55:11 +0700
Subject: [PATCH 2/3] Write merge-push.sh

---
 merge-push.sh | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100755 merge-push.sh

diff --git a/merge-push.sh b/merge-push.sh
new file mode 100755
index 0000000000..8cc428d5a0
--- /dev/null
+++ b/merge-push.sh
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# set -euo pipefail
+# IFS=$'\n\t'
+
+# Auto-abort merge conflicts, not possible when bash exits after those arise
+function abort_merge_conflicts() {
+    if [[ ! $(git ls-files -u) = "" ]]; then
+        echo "💗  Aborting merge"
+        git merge --abort
+    fi
+}
+
+function fetch_merge() {
+    remote="$1"
+    branch="$2"
+    echo "💗  Updating $branch"
+    git fetch $remote $branch:$branch
+
+    echo "💗  Merging from $remote/$branch"
+    git merge --no-edit $remote/$branch
+    abort_merge_conflicts
+}
+
+set -x
+
+git remote get-url gnu &>/dev/null ||
+    git remote add gnu https://git.savannah.gnu.org/git/emacs.git
+
+# git fetch --all
+
+upstream_main_branch="${1:-master}"
+
+fetch_merge gnu $upstream_main_branch
+
+git push --all
-- 
2.38.1


From abcaedd7e5cd67a2b8c0243a5f1e5e2e7ca2fc14 Mon Sep 17 00:00:00 2001
From: Daanturo <daanturo@gmail.com>
Date: Tue, 29 Nov 2022 13:31:36 +0700
Subject: [PATCH 3/3] Update merger-push.sh

- Make upstream_main_branch mandatory

- Allow optional second remote & branch as $2 & $3
---
 merge-push.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/merge-push.sh b/merge-push.sh
index 8cc428d5a0..705823d846 100755
--- a/merge-push.sh
+++ b/merge-push.sh
@@ -29,8 +29,10 @@
 
 # git fetch --all
 
-upstream_main_branch="${1:-master}"
+upstream_main_branch="${1}"
 
 fetch_merge gnu $upstream_main_branch
 
+[[ -n "$3" ]] && fetch_merge $2 $3
+
 git push --all
-- 
2.38.1

